cdap-api/src/main/java/co/cask/cdap/api/flow/FlowletDefinition.java

```diff
@@ -59,15 +59,10 @@
 
   public FlowletDefinition(String flowletName, Flowlet flowlet, int instances) {
     FlowletSpecification flowletSpec;
-    //TODO: CDAP-2943 Remove deprecated methods in Flow/Flowlet and move the configure methods.
-    if (flowlet instanceof AbstractFlowlet) {
-      DefaultFlowletConfigurer flowletConfigurer = new DefaultFlowletConfigurer(flowlet);
-      AbstractFlowlet abstractFlowlet = (AbstractFlowlet) flowlet;
-      abstractFlowlet.configure(flowletConfigurer);
-      flowletSpec = flowletConfigurer.createSpecification();
-    } else {
-      flowletSpec = flowlet.configure();
-    }
+    DefaultFlowletConfigurer flowletConfigurer = new DefaultFlowletConfigurer(flowlet);
+    AbstractFlowlet abstractFlowlet = (AbstractFlowlet) flowlet;
```

Comment:
shouldn't cast to AbstractFlowlet, since it's technically possible for somebody to implement Flowlet directly.  Also, the configure(FlowletConfigurer) method is in Flowlet so don't need to cast.